### ISSUE TITLE
Support CSS-in-JS in script-tag

### ISSUE THREAD

**Issue by SimonSimCity** [stylelint/stylelint#3893]
As I was asked (https://github.com/stylelint/stylelint/issues/3807#issuecomment-454314614), here the feature request as a follow-up to https://github.com/stylelint/stylelint/issues/3264.

CSS-in-JS currently does not seem to be supported if your JS is in a `script`-tag.

Here's the sample code:
```js
import styled from 'styled-components';
const Text = styled.div`
  color: white,
  background: black
`;
```

When putting this in a `.js` file it works, but if I put it into the `script` tag of an `.html` file, it doesn't show the errors.

It can also be proven by following https://github.com/stylelint/stylelint/issues/3807#issuecomment-454293030 as long as my code there still falsely is reported as being wrong.

**Comment by SimonSimCity** [comment]
@jeddy3 Just to ask this here first ... isn't the error here also wrongly marked as error..? Or am I missing out on something? See https://hackernoon.com/all-you-need-to-know-about-css-in-js-984a72d48ebc#a4dd

**Comment by SimonSimCity** [comment]
There's no real use-case here from my side ... it was just the first snippet code I could confirm gave a different result in an `html` file wrapped into a `script` tag than in a `js` file and was based on a library you used to verify https://github.com/stylelint/stylelint/issues/3264#issuecomment-390000339

Sorry for the confusion with the second comment. I guess the tutorial I used there was off or for an old version. The sample I provided in the first comment contains some errors which should (at least in my opinion) be visible in both - the `js` file and the `script` tag inside the `html` file, since both is javascript.

**Comment by gucong3000** [comment]
> I suspect extracting CSS from CSS-in-JS containers within <style> tag containers would be quite a bit more work

It won't be difficult, I just didn't develop this feature because I was worried that there was no real use case.

If someone needs this feature in a real project, I will develop it.
